wagtail/wagtail
# Allow `ModelViewSet` to be used for models with non-integer primary keys

Similar fixes to #4166 and #7208.

To test:

I don't have a bakerydemo branch for this, but I've been using the testapp. You can configure this in the following:

- Edit `DEBUG` in `wagtail/test/settings.py` to `True` (would love to change this in a separate PR)
- Run:
  ```shell
  cd wagtail/test
  DATABASE_NAME=wagtail.db python manage.py migrate
  DATABASE_NAME=wagtail.db python manage.py createcachetable
  DATABASE_NAME=wagtail.db python manage.py loaddata test.json # optional
  DATABASE_NAME=wagtail.db python manage.py loaddata test_specific.json # optional
  DATABASE_NAME=wagtail.db python manage.py createsuperuser
  DATABASE_NAME=wagtail.db python manage.py runserver
  ```

The server will run with the testapp configured. You can then use the Feature Complete Toy model to test the URLs. For testing the legacy redirects in `ModelViewSet`, you need to use a primary key string that's also a valid integer, as the old URL patterns use `<int:pk>`.

_Please check the following:_


[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)
